include/NovelRT/Experimental/Input/Glfw/GlfwInputDevice.h

```diff
@@ -0,0 +1,30 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#ifndef NOVELRT_GLFWINPUTDEVICE_H
+#define NOVELRT_GLFWINPUTDEVICE_H
+
+#ifndef NOVELRT_EXPERIMENTAL_INPUT_GLFW_H
+#error NovelRT does not support including types explicitly by default. Please include Input.Glfw.h instead for the Input::Glfw namespace subset.
+#endif
+
+namespace NovelRT::Experimental::Input::Glfw
+{
+    class GlfwInputDevice : public IInputDevice
+    {
+    private:
+        void processKeyStates() final;
+        void processKeyState(KeyCode code, KeyState state) final;
+        void acceptMouseButtonClickPush(int32_t button,
+                                        int32_t action,
+                                        Maths::GeoVector2F mousePosition) final;
+        void acceptKeyboardInputBindingPush(int32_t key, int32_t action) final;
+
+    public:
+        GlfwInputDevice() noexcept;
+
+        void consumePlayerInput() final;
```

Comment:
We use PascalCase for methods. Anything camelCase is of the old convention that we've yeeted out the window.

With that said, I'm not sure we want to be tying our input support to just mouse and keyboard. I think there's a lot to be discussed here, and a big flaw of the old system was that we basically only listened for mouse and keyboard, and while we probably will only support mouse and keyboard at first, the setup should be flexible enough to support, say, `XInput` for example.

I do think therer needs to be a proper discussion on how we want to approach this, just voicing my thoughts.